driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/TestSubscriber.java

```diff
@@ -165,7 +166,7 @@ public Subscription getSubscription() {
      * @throws AssertionError if the sequence of items observed does not exactly match {@code items}
      */
     public void assertReceivedOnNext(final List<T> items) {
-        if (getOnNextEvents().size() != items.size()) {
+        if (tryMultipleTimes(4, () -> getOnNextEvents().size() != items.size())) {
```

Comment:
I see this method called in this pattern a lot: 

        subscriber.assertNoErrors()
        subscriber.assertTerminalEvent()
        subscriber.assertReceivedOnNext([1])

I'm not clear on why we don't need the same treatment for the other assertions.

Reply:
We are seeing an error where the size comparison fails and then the error message shows they have the same size - its indicative of the race condition here.

Reply:
Are we seeing it in places where we successfully asserted the terminal event?

Reply:
Not normally - mainly change streams